tests/e2e/checker/rpcchecker.go

```diff
@@ -0,0 +1,259 @@
+	ecYaml "github.com/cilium/tetragon/api/v1/tetragon/codegen/eventchecker/yaml"
+	"github.com/cilium/tetragon/api/v1/tetragon/codegen/helpers"
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/tests/e2e/state"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/yaml"
+)
+
+// RPCChecker checks gRPC events from one or more events streams.
+type RPCChecker struct {
+	name             string
+	checker          ec.MultiEventChecker
+	getEvents        *ClientMultiplexer
+	eventLimit       uint32
+	timeLimit        time.Duration
+	logs             *bytes.Buffer
+	checkerStartedWG *sync.WaitGroup
+	lock             *sync.Mutex
+}
+
+// NewRPCChecker constructs a new RPCChecker from a MultiEventChecker.
+func NewRPCChecker(checker ec.MultiEventChecker, name string) *RPCChecker {
+	rc := &RPCChecker{
+		name:             name,
+		checker:          checker,
+		getEvents:        nil,
+		eventLimit:       0,
+		timeLimit:        0,
+		logs:             new(bytes.Buffer),
+		checkerStartedWG: new(sync.WaitGroup),
+		lock:             new(sync.Mutex),
+	}
+	// Mark that the checker has not yet started.
+	rc.checkerStartedWG.Add(1)
+	return rc
+}
+
+// WithEventLimit sets the event limit for an RPCChecker. If RPCChecker.Check sees more
+// events than the limit, it returns an error.
+func (rc *RPCChecker) WithEventLimit(limit uint32) *RPCChecker {
+	rc.eventLimit = limit
+	return rc
+}
+
+// WithTimeLimit sets the time limit for an RPCChecker. RPCChecker.Check takes longer
+// than the time limit, it returns an error.
+func (rc *RPCChecker) WithTimeLimit(limit time.Duration) *RPCChecker {
+	rc.timeLimit = limit
+	return rc
+}
+
+// Wait returns a features.Func that waits for the checker to have started its checks
+// or fails the test on timeout.
+func (rc *RPCChecker) Wait(waitTimeout time.Duration) features.Func {
+	return func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		if !cfg.ParallelTestEnabled() {
+			assert.Fail(t, "rpc checker requires parallel tests to be enabled")
+			return ctx
+		}
+		c := make(chan struct{})
+		go func() {
+			defer close(c)
+			klog.V(2).InfoS("Waiting for checker to start...", "checker", rc.name)
+			rc.checkerStartedWG.Wait()
+			klog.V(2).InfoS("Done waiting for checker to start", "checker", rc.name)
+		}()
+		select {
+		case <-time.After(waitTimeout):
+			assert.Fail(t, fmt.Sprintf("failed to wait for checker %s to start after %s", rc.name, waitTimeout))
+			return ctx
+		case <-c:
+			return ctx
+		}
+	}
+}
+
+func (rc *RPCChecker) Check(connTimeout time.Duration) features.Func {
+	return rc.CheckWithFilters(connTimeout, []*tetragon.Filter{}, []*tetragon.Filter{})
+}
+
+func (rc *RPCChecker) CheckNamespace(connTimeout time.Duration, namespaces ...string) features.Func {
```

Comment:
Maybe rename to CheckWithNamespaceFilters ?

Reply:
How about CheckInNamespace? I don't want it to be too verbose.

Reply:
Perfect!